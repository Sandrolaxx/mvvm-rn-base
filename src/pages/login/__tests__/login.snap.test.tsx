import renderer from "react-test-renderer";
import { LoginViewModel } from "../types";
import LoginView from "../view";

function mockGetFnLoginViewModel(email: string, pass: string, loading: boolean) {
    const fnLoginViewModel = jest.fn();

    fnLoginViewModel.mockImplementation(
        () =>
        ({
            email: email,
            isLoading: loading,
            password: pass,
            setEmail: jest.fn(),
            onSubmit: jest.fn(),
            setPassword: jest.fn()
        } as LoginViewModel)
    );

    return fnLoginViewModel;
}

jest.mock("../view.model", () => {
    return mockGetFnLoginViewModel("", "", false);
});

it("Should render correctly", () => {
    const { toJSON } = renderer.create(<LoginView />);

    expect(toJSON()).toMatchSnapshot();
});

jest.mock("../view.model", () => {
    return mockGetFnLoginViewModel("sandrolax@gmail.com", "234234", false);
});

it("Should render correctly with email and password", () => {
    const { toJSON } = renderer.create(<LoginView />);

    expect(toJSON()).toMatchSnapshot();
});

jest.mock("../view.model", () => {
    return mockGetFnLoginViewModel("sandrolax@gmail.com", "234234", true);
});

it("Should render correctly while is loading", () => {
    const { toJSON } = renderer.create(<LoginView />);

    expect(toJSON()).toMatchSnapshot();
});