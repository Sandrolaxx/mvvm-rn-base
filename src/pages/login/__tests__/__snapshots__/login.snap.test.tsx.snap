// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@joao.com"
      value="sandrolax@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      value="234234"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Should render correctly while is loading 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@joao.com"
      value="sandrolax@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      value="234234"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Should render correctly with email and password 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="email@joao.com"
      value="sandrolax@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="*****"
      value="234234"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
